salt/modules/status.py

```diff
@@ -366,8 +366,12 @@ def nproc():
 
         salt '*' status.nproc
     '''
-    data = __salt__['cmd.run']('nproc')
+    cmd = {
+        'Linux': 'nproc',
+        'FreeBSD': 'sysctl -n hw.ncpu',
+    }
     try:
+        data = __salt__['cmd.run'](cmd[__grains__['kernel']])
```

Comment:
cool, so if I understand you well, this code runs in grains, and thus you want to merge it as is .. otherwise let me know how to improve it